### PR TITLE
Fix/UI fix

### PR DESCRIPTION
**Pull request by Kapestian**
# UI fix
I think It is pointless to write a novel about this PR as it mostly brings visual changes. The only new features are two buttons to the AI discussion starter, one to re-generate a discussion starter and the other to copy the text to clipboard (which gives more flexibility to the user). I also added a backhandler to the account creation screen to prevent backward navigation, which causes permanent undefined behavior. Note that this screen is still vulnerable to the event of the app being closed.

Here is a [Google Slide ](https://docs.google.com/presentation/d/1We7XQx4jNH_R21lWweS9BOBxAAkzqdfOqa-CrfPBoJU/edit?usp=sharing) that recaps the primary changes.
> [!important]
Dear reviewer, in addition to the usual check of my code's quality, I need you to give me your feedback on the new design. I still have a bit of time to do changes

> [!note]
The PR is lacking coverage. This is due to the fact that I edited uncovered code, but also that some of my previous code's tests were not taken into account by sonar cloud (this was already a problem last time).

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt [hunk 1]
```diff
@@ -259,11 +270,35 @@ fun BottomSheet(aiState: AiViewModel.UiState, onDismissRequest: () -> Unit) {
...
                 is AiViewModel.UiState.Success -> {
                   Text(
-                      text = "\"${aiState.data}\"",
+                      text = "«${aiState.data}»",
                       fontWeight = FontWeight.Normal,
                       fontSize = TextUnit(CONTENT_FONT_SIZE, TextUnitType.Sp),
                       lineHeight = TextUnit(CONTENT_LINE_HEIGHT, TextUnitType.Sp),
                       modifier = Modifier.testTag("aiResponse"))
+
+                  Spacer(modifier = Modifier.height(30.dp))
```
Comment (JanStaszewiczEPFL) [inline app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:279]: That's a magic number there my friend!

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt [hunk 2]
```diff
@@ -259,11 +270,35 @@ fun BottomSheet(aiState: AiViewModel.UiState, onDismissRequest: () -> Unit) {
...
+                      horizontalArrangement = Arrangement.Center,
+                      modifier = Modifier.fillMaxWidth()) {
+                        Button(
+                            onClick = onAiRetry,
+                            elevation = ButtonDefaults.buttonElevation(BUTTONS_ELEVATION),
+                            modifier = Modifier.testTag("aiRetry")) {
+                              Icon(imageVector = Icons.Filled.Refresh, contentDescription = "Retry")
+                            }
+
+                        Spacer(modifier = Modifier.width(20.dp))
```
Comment (JanStaszewiczEPFL) [inline app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:291]: Again, a magic number! No magic allowed here! 😠

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt [hunk 3]
```diff
@@ -107,13 +115,8 @@ fun InfoSection(profile: Profile, tagsViewModel: TagsViewModel) {
...
               Text(
                   text = "Description",
-                  style =
-                      TextStyle(
-                          fontSize = 14.sp,
-                          lineHeight = 20.sp,
-                          fontWeight = FontWeight.W500,
-                          color = MaterialTheme.colorScheme.secondary,
-                          letterSpacing = 0.1.sp))
+                  style = INFO_SUBTITLE_STYLE,
```
Comment (JanStaszewiczEPFL) [inline app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt:118]: Wow I didn't know that you can use styling in values like that, nice to see!

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt [hunk 4]
```diff
@@ -259,11 +270,35 @@ fun BottomSheet(aiState: AiViewModel.UiState, onDismissRequest: () -> Unit) {
...
+                  Spacer(modifier = Modifier.height(30.dp))
+
+                  Row(
+                      horizontalArrangement = Arrangement.Center,
+                      modifier = Modifier.fillMaxWidth()) {
+                        Button(
+                            onClick = onAiRetry,
+                            elevation = ButtonDefaults.buttonElevation(BUTTONS_ELEVATION),
+                            modifier = Modifier.testTag("aiRetry")) {
+                              Icon(imageVector = Icons.Filled.Refresh, contentDescription = "Retry")
```
Comment (JanStaszewiczEPFL) [inline app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:288]: Maybe we can put the contentDescription in a string.xml value?

**Review by JanStaszewiczEPFL — CHANGES_REQUESTED** [review]
Thanks for these UI improvements, it is way better like that. I just found a few magic numbers and maybe some hardcoded strings. Otherwise great code buddy!

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt (hunk 4 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:288]: you are right, I'll do that for both buttons

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt (hunk 1 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:279]: After this project, these two words will definitely trigger a PTSD

File: app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt (hunk 2 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/profile/OtherProfileView.kt:291]: pandasproblems-the-office
I hope it's the last one of my career

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt (hunk 3 above)
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/shared/DisplayProfile.kt:118]: Yes ! I wished we had more time so we could have centralized all text styles in a single class for a more consiistent UI

**Review by JanStaszewiczEPFL — APPROVED** [review]
LGTM

**Pull request merged**